instrumentation/net/http/otelhttp/handler.go

```diff
@@ -281,3 +283,34 @@ func WithRouteTag(route string, h http.Handler) http.Handler {
 		h.ServeHTTP(w, r)
 	})
 }
+
+func withoutCancel(ctx context.Context) context.Context {
+	if ctx.Done() == nil {
+		return ctx
+	}
+	return withoutCancelCtx{ctx}
```

Comment:
Why not?

```suggestion
	if parent == nil {
		panic("cannot create context from nil parent")
	}
	return withoutCancelCtx{parent}
```

It is copied from https://cs.opensource.google/go/go/+/refs/tags/go1.21.4:src/context/context.go;l=563-571 therefore I would keep it.

Reply:
Sounds good, I didn't know this had been copied from this. The thinking was that, if a context that wasn't cancellable was passed to `withoutCancel` it seemed like we could return/no-op (and, if it was `nil`, we'd get a panic).